crates/nargo/src/cli/mod.rs

```diff
@@ -143,7 +155,15 @@ fn write_inputs_to_file<P: AsRef<Path>>(
         dir_path
     };
 
-    let serialized_output = format.serialize(w_map)?;
+    // If it exists, insert return value into input map so it's written to file.
+    let serialized_output = if let Some(return_value) = return_value {
```

Comment:
can we use a match here, for ergonomics?

Reply:
Sure thing, addressed in [3942386](https://github.com/noir-lang/noir/pull/885/commits/3942386c9e9b196057540c578cf4c6f89d706bdb)